### PR TITLE
Feature/jstassi/mk restarts on sles12

### PR DESCRIPTION
**Pull request by gmao-jstassi**
The mk_Restarts script was modified to run on SLES-12. This change was made in conjunction with updates in regrid.pl in the @GMAO_Shared component. I ran tests that were successful on both SLES-11 and SLES-12.

### CODE REVIEW

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSland_GridComp/GEOScatch_GridComp/mk_restarts/mk_Restarts [hunk 1]
```diff
@@ -214,7 +215,7 @@ sub write_mk_catchcn_j {
 #SBATCH --time=$walltime
 #SBATCH --job-name=catchcnj
 #SBATCH --constraint=hasw
```
Comment (sdrabenh) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSland_GridComp/GEOScatch_GridComp/mk_restarts/mk_Restarts:217]: This should be changed to allow sky as well

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSland_GridComp/GEOScatch_GridComp/mk_restarts/mk_Restarts [hunk 2]
```diff
@@ -309,7 +310,9 @@ sub wait_for_pid {
     $first = 1;
     while (1) {
         %found = ();
-        foreach $line (`qstat | grep $ENV{"USER"}`) {
+        #--foreach $line (`qstat | grep $ENV{"USER"}`) {
+        foreach $line (`squeue | grep $ENV{"USER"}`) {
```
Comment (sdrabenh) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSland_GridComp/GEOScatch_GridComp/mk_restarts/mk_Restarts:314]: Will this fail running on PBS at NAS?

**Comment by sdrabenh** [comment]
@gmao-jstassi thanks for working on this. Your first PR in git ;-) Please see my comments above as they may cause potential problems down the road. I ran this on SLES12 Haswell and most of it worked fine. However, I ran into a seg fault issues regridding the bkg_eta_rst (see errors below). This seems to run fin on SLES11, but not SLES12:

```====================
regrid bkg03_eta_rst
====================
move ./C360CS_f522_fp.bkg03_eta_rst.20190614_21z.nc4
  => ./C360CS_f522_fp.bkg03_eta_rst.20190614_21z.nc4.orig
/gpfsm/dnb78s1/projects/p18/sdrabenh/Models/GIT_PRs/PRs_sles12_rst/install-sles12/bin/dyn2dyn.x -g5 -res d -nlevs 72 -o /gpfsm/dnb78s1/projects/p18/sdrabenh/regrid/x0041/Summer-sles12/C360CS_f522_fp.bkg03_eta_rst.20190614_21z.nc4 /gpfsm/dnb78s1/projects/p18/sdrabenh/regrid/x0041/Summer-sles12/C360CS_f522_fp.bkg03_eta_rst.20190614_21z.nc4.orig
 
 ------------------------------------------------------------------
   Eta     Dynamics state files:            1
                                 
 /gpfsm/dnb78s1/projects/p18/sdrabenh/regrid/x0041/Summer-sles12/C360CS_f522_fp.
 bkg03_eta_rst.20190614_21z.nc4.orig
 GFIO_GetRealAtt: error reading attribute info
 GFIO_GetIntAtt: error reading attribute info
 m_dyn2dyn::dyn2dyn_do1_: Horizontally interpolating from         1152 x
         721  to          576 x         361
  Treating winds as on A-grid 
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source             
dyn2dyn.x          00000000004F816D  Unknown               Unknown  Unknown
libpthread-2.22.s  00007FFFF25C7C10  Unknown               Unknown  Unknown
dyn2dyn.x          000000000046644D  m_maph_mp_ymap_           797  m_maph.F90
dyn2dyn.x          0000000000466306  m_maph_mp_map_a2a         737  m_maph.F90
dyn2dyn.x          00000000004627DA  m_maph_mp_h_map_d         480  m_maph.F90
dyn2dyn.x          00000000004613A2  m_maph_mp_h_map_d         239  m_maph.F90
dyn2dyn.x          000000000045DE02  m_dyn2dyn_mp_dyn2         373  m_dyn2dyn.f90
dyn2dyn.x          00000000004134D5  MAIN__                    160  dyn2dyn.f90
dyn2dyn.x          0000000000412A6E  Unknown               Unknown  Unknown
libc-2.22.so       00007FFFF18FA725  __libc_start_main     Unknown  Unknown
dyn2dyn.x          0000000000412969  Unknown               Unknown  Unknown
Error with /gpfsm/dnb78s1/projects/p18/sdrabenh/Models/GIT_PRs/PRs_sles12_rst/install-sles12/bin/dyn2dyn.x; at /discover/nobackup/projects/gmao/g6dev/sdrabenh/Models/GIT_PRs/PRs_sles12_rst/install/bin/regrid.pl line 3193.```

**Comment by sdrabenh** [comment]
@mathomp4 any idea what may have caused this seg fault in SLES12? I ran this exact same thing in SLES11 and there were no problems.

**Comment by mathomp4** [comment]
> @mathomp4 any idea what may have caused this seg fault in SLES12? I ran this exact same thing in SLES11 and there were no problems.

@sdrabenh I think that is more in @bena-nasa or @atrayano territory. Can you send in a ticket to our system with how you were running regrid so we can try to duplicate.

**Comment by gmao-jstassi** [comment]
Hi Scott,

I made a couple of comments on the github.com page for mk_Restarts, but I couldn't figure out how to respond to this issue on the Web page.

I just wanted to say that I have run a couple of tests on SLES-12 with my script updates, tests which include running dyn2dyn.x for bkg_eta_rst files, and I am not having this problem. The tests look good. Could this be an environment issue? I am pretty sure that I am using a different compiler than you.

Joe
________________________________

**Pull request merged**